cmd/nerdctl/main_linux.go

```diff
@@ -66,3 +69,21 @@ func appNeedsRootlessParentMain(cmd *cobra.Command, args []string) bool {
 func addApparmorCommand(rootCmd *cobra.Command) {
 	rootCmd.AddCommand(apparmor.Command())
 }
+
+var (
+	getuid    = unix.Getuid
+	geteuid   = unix.Geteuid
+	setresuid = unix.Setresuid
+)
+
+func dropPrivileges() error {
+	var err error
+	if runtime.GOOS == "linux" {
+		uid := getuid()
+		euid := geteuid()
+		if uid != euid && euid == 0 {
```

Comment:
what happens if we `setuid` for non-root user ?

Reply:
Who does that, for what purpose?